### PR TITLE
Add com.warlordsoftwares.formatlab

### PR DESCRIPTION
**Pull request by rishabh3354**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by rishabh3354** [comment]
bot, build com.warlordsoftwares.formatlab

File: com.warlordsoftwares.formatlab.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - mkdir -p ${FLATPAK_DEST}/formatlab
+      - tar -xzf *latest.tar.gz -C ${FLATPAK_DEST}/formatlab
+      - install -Dm755 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/formatlab.sh ${FLATPAK_DEST}/bin/formatlab.sh
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/icons/icon.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/com.warlordsoftwares.formatlab.png
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/applications/formatlab.desktop ${FLATPAK_DEST}/share/applications/com.warlordsoftwares.formatlab.desktop
+      - desktop-file-edit --set-icon=${FLATPAK_ID} ${FLATPAK_DEST}/share/applications/com.warlordsoftwares.formatlab.desktop
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/metainfo/com.warlordsoftwares.formatlab.metainfo.xml ${FLATPAK_DEST}/share/metainfo/com.warlordsoftwares.formatlab.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/rishabh3354/FLATPAK_FORMAT_LAB/archive/refs/tags/latest.tar.gz
```
Comment (hfiguiere) [inline com.warlordsoftwares.formatlab.yml:72]: this should be a specific commit or tag. `latest` mean that next time it's update the build will break, and no one will be able to reproduce the previous built.

File: com.warlordsoftwares.formatlab.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/formatlab
+      - tar -xzf *latest.tar.gz -C ${FLATPAK_DEST}/formatlab
+      - install -Dm755 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/formatlab.sh ${FLATPAK_DEST}/bin/formatlab.sh
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/icons/icon.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/com.warlordsoftwares.formatlab.png
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/applications/formatlab.desktop ${FLATPAK_DEST}/share/applications/com.warlordsoftwares.formatlab.desktop
+      - desktop-file-edit --set-icon=${FLATPAK_ID} ${FLATPAK_DEST}/share/applications/com.warlordsoftwares.formatlab.desktop
+      - install -Dm644 ${FLATPAK_DEST}/formatlab/FLATPAK_FORMAT_LAB-latest/share/metainfo/com.warlordsoftwares.formatlab.metainfo.xml ${FLATPAK_DEST}/share/metainfo/com.warlordsoftwares.formatlab.metainfo.xml
+    sources:
+      - type: file
```
Comment (hfiguiere) [inline com.warlordsoftwares.formatlab.yml:71]: this should be type `archive`, and properly install things where they should be.

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: since it's a python app, why this limitation?

Comment (rishabh3354) [reply flathub.json]: Thanks for the information. i have removed this limitation.

File: com.warlordsoftwares.formatlab.yml (hunk 2 above)
Comment (rishabh3354) [reply com.warlordsoftwares.formatlab.yml]: got it. Thanks. It is now resolved!

File: com.warlordsoftwares.formatlab.yml (hunk 1 above)
Comment (rishabh3354) [reply com.warlordsoftwares.formatlab.yml]: Got it, It is now resolved!

**Comment by rishabh3354** [comment]
bot, build com.warlordsoftwares.formatlab

**Comment by rishabh3354** [comment]
bot, build com.warlordsoftwares.formatlab

**Pull request closed without merging**